R/assocTestSingle.R

```diff
@@ -15,6 +15,18 @@ setMethod("assocTestSingle",
               # Convert old null model format if necessary.
               null.model <- .updateNullModelFormat(null.model)
 
+              # check null.model for needed components
+              calc.score <- test %in% c("Score", "Score.SPA") | (recalc.pval.thresh < 1)
+              if(calc.score && approx.score.SE && is.null(null.model$score.se.ratio)){
+              	stop("null.model must have score.se.ratio when approx.score.SE = TRUE")
+              } 
+              if(calc.score && !(approx.score.SE) && isNullModelSmall(null.model)){
+              	stop("small null.model cannot be used when approx.score.SE = FALSE")
+              }
+              if(!is.null(GxE) && isNullModelSmall(nullmod)){
+              	stop("small null.model cannot be used with GxE")
+              }
+              
```

Comment:
You could turn this into an internal function (`.checkNullModelApproxScoreSE`?) -- it's easier to test.

Reply:
done.